### PR TITLE
Update some HLT tests for python3

### PR DESCRIPTION
**Pull request by davidlange6**
HLTrigger/Configuration/test/runAll.csh Fake appears to work after these changes.

### CODE REVIEW

**Comment by davidlange6** [comment]
please test
@Martin-Grunewald - do the relevant HLT tests run in the PR tests?

**Comment by Martin-Grunewald** [comment]
Unfortunately the (failing) tests are not run in PR tests, but in IB tests.

**Comment by Martin-Grunewald** [comment]
Including this PR in  CMSSW_12_0_X_2021-07-19-2300, there appears the following SyntaxWarning:
```
src/HLTrigger/Configuration/python/Tools/options.py:135: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if name is 'menu' and not isinstance(value, ConnectionHLTMenu):
src/HLTrigger/Configuration/python/Tools/options.py:138: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif name is 'l1' and not isinstance(value, ConnectionL1TMenu):
src/HLTrigger/Configuration/python/Tools/options.py:141: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif name is 'l1Xml' and not isinstance(value, ConnectionL1TMenuXml):
src/HLTrigger/Configuration/python/Tools/options.py:144: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif name is 'open' and value:
src/HLTrigger/Configuration/python/Tools/options.py:148: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif name is 'prescale' and value is not None:
src/HLTrigger/Configuration/python/Tools/options.py:152: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif name is 'profiling' and value:
src/HLTrigger/Configuration/python/Tools/options.py:155: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif name is 'timing' and value:
>> All python modules compiled
```

**Comment by davidlange6** [comment]
Right - Those are pre-existing bugs (most likely anyway) that python3 now shows you (instead which python2 ignored them)

File: FWCore/ParameterSet/scripts/edmConfigDump [hunk 1]
```diff
@@ -19,7 +20,10 @@ parser.add_argument("--pruneVerbose", action="store_true",
 
 options = parser.parse_args()
 handle = open(options.filename, 'r')
-cfo = imp.load_source("pycfg", options.filename, handle)
+loader=SourceFileLoader("pycfg", options.filename)
+loaded = types.ModuleType("pycfg",loader.name)
+loader.exec_module(loaded)
+#cfo = imp.load_source("pycfg", options.filename, handle)
```
Comment (makortel) [inline FWCore/ParameterSet/scripts/edmConfigDump:26]: Is the commented out line still useful? It seems to me that `handle` becomes now unnecessary.

**Comment by davidlange6** [comment]
> On Jul 20, 2021, at 4:08 PM, Matti Kortelainen ***@***.***> wrote:
> 
> 
> @makortel commented on this pull request.
> 
> In FWCore/ParameterSet/scripts/edmConfigDump:
> 
> > @@ -19,7 +20,10 @@ parser.add_argument("--pruneVerbose", action="store_true",
>  
>  options = parser.parse_args()
>  handle = open(options.filename, 'r')
> -cfo = imp.load_source("pycfg", options.filename, handle)
> +loader=SourceFileLoader("pycfg", options.filename)
> +loaded = types.ModuleType("pycfg",loader.name)
> +loader.exec_module(loaded)
> +#cfo = imp.load_source("pycfg", options.filename, handle)
> 
> Is the commented out line still useful? It seems to me that handle becomes now unnecessary.

Yes that can be cleaned up. Thanks for noticing that

File: FWCore/ParameterSet/scripts/edmConfigDump [hunk 2]
```diff
@@ -19,7 +20,10 @@ parser.add_argument("--pruneVerbose", action="store_true",
...
 cmsProcess = cfo.process
```
Comment (Martin-Grunewald) [inline FWCore/ParameterSet/scripts/edmConfigDump:27]: This line crashes as ```cfo``` is no longer defined.

**Comment by davidlange6** [comment]
Ha - that is obviously correct. What test gave you that error?

**Comment by Martin-Grunewald** [comment]
runAll.csh Fake
Even though the script says exit code 0, the log file displays the error.

**Comment by Martin-Grunewald** [comment]
But more straight forward, simply:
```
edmConfigDump OnLine_HLT_Fake.py
Traceback (most recent call last):
  File "/data/CMS0/CMSSW_12_0_X_2021-07-19-2300/bin/slc7_amd64_gcc10/edmConfigDump", line 27, in <module>
    cmsProcess = cfo.process
NameError: name 'cfo' is not defined
```

**Comment by davidlange6** [comment]
ok, edmConfigDump <an_hlt_config.py> works now

**Comment by davidlange6** [comment]
yea, thats what fooled me..

>Even though the script says exit code 0, the log file of the HLT_Integration test run as a part of it displays the error.

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
+1

**Comment by Martin-Grunewald** [comment]
urgent

**Comment by Martin-Grunewald** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**